components/frontend/src/utils.js

```diff
@@ -96,6 +96,16 @@ export function getMetricUnit(metric, dataModel) {
     return metric.unit || dataModel.metrics[metric.type].unit || ""
 }
 
+export function isMeasurementRequested(metric) {
+    if (metric.measurement_requested) {
+        if (metric.latest_measurement) {
+            return new Date(metric.measurement_requested) >= new Date(metric.latest_measurement.end)
```

Comment:
Does this automatically take timezone differences between server and client into account?

Reply:
Oh, good one. Let me check. Yes, the frontend uses `new Date().toISOString()` which converts the date to an ISO-formatted string in UTC.